hadoop/src/main/java/org/apache/carbondata/hadoop/util/Hive2CarbonExpression.java

```diff
@@ -0,0 +1,177 @@
+            getDateType(ll.get(left).getTypeString()));
+        return new NotEqualsExpression(columnExpression, literalExpression);
+      } else if (udf instanceof GenericUDFOPEqualOrLessThan) {
+        ColumnExpression columnExpression = new ColumnExpression(ll.get(left).getCols().get(left),
+            getDateType(ll.get(left).getTypeString()));
+        LiteralExpression literalExpression = new LiteralExpression(ll.get(right).getExprString(),
+            getDateType(ll.get(left).getTypeString()));
+        return new LessThanEqualToExpression(columnExpression, literalExpression);
+      } else if (udf instanceof GenericUDFOPLessThan) {
+        ColumnExpression columnExpression = new ColumnExpression(ll.get(left).getCols().get(left),
+            getDateType(ll.get(left).getTypeString()));
+        LiteralExpression literalExpression = new LiteralExpression(ll.get(right).getExprString(),
+            getDateType(ll.get(left).getTypeString()));
+        return new LessThanExpression(columnExpression, literalExpression);
+      } else if (udf instanceof GenericUDFOPNull) {
+        ColumnExpression columnExpression = new ColumnExpression(ll.get(left).getCols().get(left),
+            getDateType(ll.get(left).getTypeString()));
+        LiteralExpression literalExpression = new LiteralExpression(null, null);
+        return new EqualToExpression(columnExpression, literalExpression, true);
+      } else if (udf instanceof GenericUDFOPNotNull) {
+        ColumnExpression columnExpression = new ColumnExpression(ll.get(left).getCols().get(left),
+            getDateType(ll.get(left).getTypeString()));
+        LiteralExpression literalExpression = new LiteralExpression(null, null);
+        return new NotEqualsExpression(columnExpression, literalExpression, true);
+      } else {
+        LOG.error("error:not find type" + udf.toString());
+      }
+    }
+    return null;
+  }
+
+  public static DataType getDateType(String type) {
+    if (type.toUpperCase().endsWith("INT")) {
+      return DataTypes.INT;
+    } else if (type.toUpperCase().endsWith("STRING")) {
+      return DataTypes.STRING;
+    } else if (type.toUpperCase().endsWith("LONG")) {
+      return DataTypes.LONG;
+    } else if (type.toUpperCase().endsWith("FLOAT")) {
+      return DataTypes.FLOAT;
+    } else if (type.toUpperCase().endsWith("DATE")) {
+      return DataTypes.DATE;
+    } else if (type.toUpperCase().endsWith("TIMESTAMP")) {
+      return DataTypes.TIMESTAMP;
+    } else if (type.toUpperCase().endsWith("DOUBLE")) {
+      return DataTypes.DOUBLE;
+    } else if (type.toUpperCase().endsWith("NULL")) {
+      return DataTypes.NULL;
+    } else if (type.toUpperCase().endsWith("BYTE")) {
+      return DataTypes.BYTE;
+    } else if (type.toUpperCase().endsWith("SHORT_INT")) {
+      return DataTypes.SHORT_INT;
+    } else if (type.toUpperCase().endsWith("VARCHAR")) {
+      return DataTypes.VARCHAR;
+    }
+    return null;
```

Comment:
Is hive pushdown unsupported for complex data types?

Reply:
not support